src/importlinter/domain/fields.py

```diff
@@ -117,7 +117,7 @@ class DirectImportField(Field):
     Expects raw data in the form: "mypackage.foo.importer -> mypackage.bar.imported".
     """
 
-    DIRECT_IMPORT_STRING_REGEX = re.compile(r"^([\w\.]+) -> ([\w\.]+)$")
+    DIRECT_IMPORT_STRING_REGEX = re.compile(r"^([\w\.-]+) -> ([\w\.-]+)$")
```

Comment:
I would do:

```
class DirectImportField(Field):
    """
    A field for DirectImports.
    Expects raw data in the form: "mypackage.foo.importer -> mypackage.bar.imported".
    """
    def parse(self, raw_data: Union[str, List]) -> DirectImport:
        string = StringField().parse(raw_data)
        importer, _, imported = string.partition(' -> ')
        if not (importer and imported):
            raise ValidationError('Must be in the form "package.importer -> package.imported".')
        return DirectImport(importer=Module(importer), imported=Module(imported))
```

Instead, and avoid the `re` altogether!

Reply:
I like both ways, don't feel strongly.

Reply:
```
>>> import timeit
>>> timeit.timeit('x.match("hello -> yellow").groups()', 'import re; x = re.compile(r"^([\w\.]+) -> ([\w\.]+)$")')
0.6223237819999952
>>> timeit.timeit('x.match("hello -> yellow").groups()', 'import re; x = re.compile(r"^([\w\.]+) -> ([\w\.]+)$")')
0.6110563539999987
>>> timeit.timeit('x.match("hello -> yellow").groups()', 'import re; x = re.compile(r"^([\w\.]+) -> ([\w\.]+)$")')
0.6207327580000026
>>> timeit.timeit('"hello -> yellow".partition(" -> ")')
0.1567364259999806
>>> timeit.timeit('"hello -> yellow".partition(" -> ")')
0.15910449500000823
>>> timeit.timeit('"hello -> yellow".partition(" -> ")')
0.15209861200000319
```

I like regex -- but i only use it when it's something complex.. this doesn't warren it.